pkg/graph/resolvers/mto_milestone.go

```diff
@@ -78,13 +79,61 @@ func MTOMilestoneCreateCommon(ctx context.Context, logger *zap.Logger, principal
 		}
 
 		// A common milestone never has a name (since it comes from the Common Milestone itself), so pass in `nil`
-		milestone := models.NewMTOMilestone(principalAccount.ID, nil, &commonMilestoneKey, modelPlanID, &finalCategoryID)
+		milestone := models.NewMTOMilestone(
+			principalAccount.ID,
+			nil,
+			&commonMilestoneKey,
+			modelPlanID,
+			&finalCategoryID,
+		)
 		milestone.FacilitatedBy = &commonMilestone.FacilitatedByRole
 
 		if err := BaseStructPreCreate(logger, milestone, principal, store, true); err != nil {
 			return nil, err
 		}
-		return storage.MTOMilestoneCreate(tx, logger, milestone)
+		createdMilestone, err := storage.MTOMilestoneCreate(tx, logger, milestone)
+
+		var solutions []*models.MTOSolution
+		for _, commonSolutionKey := range commonSolutions {
+			solution := models.NewMTOSolution(modelPlanID,
+				&commonSolutionKey,
+				nil,
+				nil,
+				nil,
+				principal.Account().ID)
+
+			mtoSolution, createMTOSolutionErr := storage.MTOSolutionCreateAllowConflicts(tx, logger, solution)
+			if createMTOSolutionErr != nil {
+				logger.Error("failed to create solution when creating common milestone", zap.Error(err))
+				return nil, createMTOSolutionErr
+			}
+
+			solutions = append(solutions, mtoSolution)
+		}
+
+		for _, solution := range solutions {
+			mtoMilestoneSolutionLink := models.NewMTOMilestoneSolutionLink(
+				principal.Account().ID,
+				createdMilestone.ID,
+				solution.ID,
+			)
+
+			_, milestoneSolutionLinkErr := storage.MTOMilestoneSolutionLinkCreate(
```

Comment:
same as above, consider making one store method. A `TODO` is appropriate for this PR